packages/rpc-graphql/src/resolvers/account.ts

```diff
@@ -1,141 +1,196 @@
 import { Address } from '@solana/addresses';
+import { Commitment, Slot } from '@solana/rpc-types';
 import { GraphQLResolveInfo } from 'graphql';
 
 import { RpcGraphQLContext } from '../context';
-import { AccountLoaderArgs } from '../loaders';
-import { onlyPresentFieldRequested } from './resolve-info';
-
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-function transformParsedAccountData(parsedAccountData: any) {
-    const {
-        parsed: { info: result, type: accountType },
-        program: programName,
-        programId,
-    } = parsedAccountData;
-    // Tells GraphQL which account type has been
-    // returned by the RPC.
-    result.accountType = accountType;
-    result.programId = programId;
-    // Tells GraphQL which program the returned
-    // account belongs to.
-    result.programName = programName;
-    return result;
+import { AccountLoaderValue } from '../loaders';
+import { buildAccountLoaderArgSetFromResolveInfo, onlyFieldsRequested } from './resolve-info';
+
+type Encoding = 'base58' | 'base64' | 'base64+zstd';
+type DataSlice = { length: number; offset: number };
+type EncodedData = {
+    data: string;
+    dataSlice: DataSlice | null;
+};
+
+export class EncodedAccountData {
+    base58: EncodedData[] = [];
+    base64: EncodedData[] = [];
+    'base64+zstd': EncodedData[] = [];
+
+    insert(data: string, dataSlice: DataSlice | null, encoding: Encoding) {
+        this[encoding].push({ data, dataSlice });
+    }
+
+    get(dataSlice: DataSlice | null, encoding: Encoding): string | undefined {
+        return this[encoding].find(entry => {
+            if (dataSlice === null) {
+                return entry.dataSlice === null;
+            }
+            return entry.dataSlice?.length === dataSlice.length && entry.dataSlice?.offset === dataSlice.offset;
+        })?.data;
+    }
 }
 
-export function transformLoadedAccount({
-    account,
-    address,
-    encoding = 'jsonParsed',
-}: {
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    account: any;
+export type AccountResult = {
```

Comment:
The diff here is terrible, but basically what I've done is laid out a structured object for an "account result", which contains a little dictionary of encoded-sliced data so that the `resolveAccountData` can just grab the proper slice from the result's dictionary.

Reply:
This will break down when you do the ‘fetch the aggregate range and I'll dole out the right slices to those who asked for them’ logic right? You'll have to replace this with a system that actually decodes the data, stores it keyed on the account, then slices and re-encodes it for each consumer.

That actually kind of sucks and I'm starting to regret it. Base58/64 is not _that_ bad, but if the user requests zstd compressed data and you fetch an aggregate of two slices, now you have to include a zstd compressor/decompressor _in the bundle_. And that thing is wasm/async too.

Oof. Regrets.

Reply:
Actually the batch loader handles this properly by slicing the data before resolving the promise. So, by the time data comes back to the resolver from the loader, it's got the right indices and the requested data slice arguments.

With that being said, I'll need to handle `base64+zstd` separately on the batch loader itself, so that it's always treated with an isolated request (yuck).

Regardless, this shouldn't break down since the loader does all of that slicing & dicing.